hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/s3/multipart/S3MultipartUploadAbortResponse.java

```diff
@@ -97,19 +97,20 @@ public void addToDBBatch(OMMetadataManager omMetadataManager,
       OmKeyInfo currentKeyPartInfo =
           OmKeyInfo.getFromProtobuf(partKeyInfo.getPartKeyInfo());
 
-      RepeatedOmKeyInfo repeatedOmKeyInfo =
-          omMetadataManager.getDeletedTable().get(partKeyInfo.getPartName());
-
-      repeatedOmKeyInfo = OmUtils.prepareKeyForDelete(currentKeyPartInfo,
-          repeatedOmKeyInfo, omMultipartKeyInfo.getUpdateID(), isRatisEnabled);
+      RepeatedOmKeyInfo repeatedOmKeyInfo = OmUtils.prepareKeyForDelete(
+          currentKeyPartInfo, null, omMultipartKeyInfo.getUpdateID(),
+          isRatisEnabled);
+      // multi-part key format is volumeName/bucketName/keyName/uploadId
+      String deleteKey = omMetadataManager.getOzoneDeletePathKey(
+          currentKeyPartInfo.getObjectID(), multipartKey);
```

Comment:
It should have `partNumber`/`partName`.  Object id is the same for all parts.

BTW, let's add a utility method to create multipart delete key.

Reply:
@szetszwo Each part is created with new object ID as independent upload, and combined to multipart table in repeatedomkeyinfo. ObjectId is not replaced in keyInfo for the part and remains unique.

but part number / name is not unique as user can do re-upload with same part number. If same part number, it will overwrite. And this can cause duplicate entry in deletedTable. So this is not unique.

Reply:
@sumitagrawl , thanks for the info.  Do we have a unit test showing it?  If not, could you add one?

Reply:
@szetszwo This is already covered by existing testcase and same is updated with objectId as correction as part of this PR,
https://github.com/apache/ozone/pull/4660/files#diff-c1e307469ed5f710651ae2ec86baed4d9dce85d742c1c282cb1e7d28a776b503

Reply:
@sumitagrawl , I am asking a unit test.  If there is one, which one?  I want to run it locally.  Thanks.

Reply:
@szetszwo TestS3MultipartUploadAbortResponse:testAddDBToBatchWithParts cover the scenario as discussed.

Reply:
@sumitagrawl , tried the `testAddDBToBatchWithParts` but it is not a real test -- in `createPartKeyInfo`, it creates parts random an object id but not the id generated by OM.

We should have a test that a real client uploads multipart to OM and then the parts get deleted.

Reply:
> ... in createPartKeyInfo, it creates parts random an object id but not the id generated by OM.

Indeed, the random ids could be the same since randomness does not prevent collision.